independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/BootstrapMavenContext.java

```diff
@@ -488,33 +488,28 @@ private File resolveSettingsFile(String settingsArg, String alternatePomDir, Str
         return null;
     }
 
+    // mostly a copy of `DefaultRepositorySystemSessionFactory.newRepositorySession()`
     private DefaultRepositorySystemSession newRepositorySystemSession() throws BootstrapMavenException {
-        final DefaultRepositorySystemSession session = MavenRepositorySystemUtils.newSession();
-        final Settings settings = getEffectiveSettings();
-        final List<Mirror> mirrors = settings.getMirrors();
-        if (mirrors != null && !mirrors.isEmpty()) {
-            final DefaultMirrorSelector ms = new DefaultMirrorSelector();
-            for (Mirror m : mirrors) {
-                ms.add(m.getId(), m.getUrl(), m.getLayout(), false, m.isBlocked(), m.getMirrorOf(), m.getMirrorOfLayouts());
-            }
-            session.setMirrorSelector(ms);
-        }
+        DefaultRepositorySystemSession session = MavenRepositorySystemUtils.newSession();
+        Settings settings = getEffectiveSettings();
 
-        final String localRepoPath = getLocalRepo();
-        final String[] localRepoTailPaths = getLocalRepoTail();
+        session.setCache(new DefaultRepositoryCache());
 
-        final LocalRepositoryManager head = getRepositorySystem().newLocalRepositoryManager(session,
-                new LocalRepository(localRepoPath));
-
-        if (localRepoTailPaths.length == 0) {
-            session.setLocalRepositoryManager(head);
-        } else {
-            final List<LocalRepositoryManager> tail = new ArrayList<>(localRepoTailPaths.length);
-            for (final String tailPath : localRepoTailPaths) {
-                tail.add(getRepositorySystem().newLocalRepositoryManager(session, new LocalRepository(tailPath)));
-            }
-            session.setLocalRepositoryManager(
-                    new ChainedLocalRepositoryManager(head, tail, getLocalRepoTailIgnoreAvailability()));
+        Map<Object, Object> configProps = new LinkedHashMap<>();
+        configProps.put(ConfigurationProperties.USER_AGENT, getUserAgent());
+        configProps.put(ConfigurationProperties.INTERACTIVE, settings.isInteractiveMode());
+        configProps.put("maven.startTime", new Date());
```

Comment:
Not sure we need this one.

Reply:
Yeah, we probably don't. I copied this from MIMA, because in Maven proper, the `maven.startTime` comes from the Maven startup request (not sure about the name), which we obviously don't have. But I can remove it if you wish.

Reply:
I'd probably remove it but it's not critical. We can do that later.

Reply:
Removed.